src/lib/V3DutchOrderLib.sol

```diff
@@ -65,41 +69,60 @@ struct V3DutchOutput {
     address recipient;
+    // The min amount of the curve
+    uint256 minAmount;
+    // The amount of token to change per wei change in basefee
+    uint256 adjustmentPerGweiBasefee;
 }
 
 /// @notice helpers for handling custom curve order objects
 library V3DutchOrderLib {
     using OrderInfoLib for OrderInfo;
 
-    bytes internal constant NON_LINEAR_DUTCH_ORDER_TYPE = abi.encodePacked(
+    bytes internal constant V3_DUTCH_ORDER_TYPE = abi.encodePacked(
         "V3DutchOrder(",
         "OrderInfo info,",
         "address cosigner,",
         "V3DutchInput baseInput,",
-        "V3DutchOutput[] baseOutputs)"
+        "V3DutchOutput[] baseOutputs," "uint256 gasSnapshot)"
     );
-    bytes internal constant NON_LINEAR_DUTCH_OUTPUT_TYPE = abi.encodePacked(
-        "V3DutchOutput(", "address token,", "uint256 startAmount,", "NonlinearDutchDecay curve,", "address recipient)"
+    bytes internal constant V3_DUTCH_INPUT_TYPE = abi.encodePacked(
+        "V3DutchInput(",
+        "address token,",
+        "uint256 startAmount,",
+        "NonlinearDutchDecay curve,",
+        "uint256 maxAmount,",
+        "uint256 adjustmentPerGweiBasefee)"
     );
-    bytes32 internal constant NON_LINEAR_DUTCH_OUTPUT_TYPE_HASH = keccak256(NON_LINEAR_DUTCH_OUTPUT_TYPE);
+    bytes32 internal constant V3_DUTCH_INPUT_TYPE_HASH = keccak256(V3_DUTCH_INPUT_TYPE);
+    bytes internal constant V3_DUTCH_OUTPUT_TYPE = abi.encodePacked(
+        "V3DutchOutput(",
+        "address token,",
+        "uint256 startAmount,",
+        "NonlinearDutchDecay curve,",
+        "address recipient,",
+        "uint256 minAmount,",
+        "uint256 adjustmentPerGweiBasefee)"
+    );
+    bytes32 internal constant V3_DUTCH_OUTPUT_TYPE_HASH = keccak256(V3_DUTCH_OUTPUT_TYPE);
     bytes internal constant NON_LINEAR_DECAY_TYPE =
         abi.encodePacked("NonlinearDutchDecay(", "uint256 relativeBlocks,", "int256[] relativeAmounts)");
     bytes32 internal constant NON_LINEAR_DECAY_TYPE_HASH = keccak256(NON_LINEAR_DECAY_TYPE);
 
-    bytes internal constant ORDER_TYPE = abi.encodePacked(
-        NON_LINEAR_DECAY_TYPE, NON_LINEAR_DUTCH_ORDER_TYPE, NON_LINEAR_DUTCH_OUTPUT_TYPE, OrderInfoLib.ORDER_INFO_TYPE
-    );
+    bytes internal constant ORDER_TYPE =
+        abi.encodePacked(NON_LINEAR_DECAY_TYPE, V3_DUTCH_ORDER_TYPE, V3_DUTCH_OUTPUT_TYPE, OrderInfoLib.ORDER_INFO_TYPE);
     bytes32 internal constant ORDER_TYPE_HASH = keccak256(ORDER_TYPE);
 
     /// @dev Note that sub-structs have to be defined in alphabetical order in the EIP-712 spec
     string internal constant PERMIT2_ORDER_TYPE = string(
         abi.encodePacked(
             "V3DutchOrder witness)",
             NON_LINEAR_DECAY_TYPE,
-            NON_LINEAR_DUTCH_ORDER_TYPE,
-            NON_LINEAR_DUTCH_OUTPUT_TYPE,
             OrderInfoLib.ORDER_INFO_TYPE,
-            DutchOrderLib.TOKEN_PERMISSIONS_TYPE
+            DutchOrderLib.TOKEN_PERMISSIONS_TYPE,
+            V3_DUTCH_INPUT_TYPE,
```

Comment:
Some of these are fixes from this PR
https://github.com/Uniswap/UniswapX/pull/269